### ISSUE TITLE
S4660 (selector-pseudo-element-no-unknown): avoid FP on ::part() pseudo selector

### ISSUE THREAD

**Issue by b0ndt** [SonarSource/sonar-css#262]
In our project we use Ionic 5 which introduced a way to style Shadow Dom elements by supporting the [Shadow Parts](https://www.w3.org/TR/css-shadow-parts-1/) standard.

As you can see [here](https://github.com/ionic-team/ionic-framework/blob/a625c837a60abc07ad71c696196a89f1a25a4c27/core/src/components/select/select.tsx#L476) the ion-select component has been extended by a part="icon" attribute on its icon Dom node. By applying properties to the following css selector, a customization is possible:

```
ion-select::part(icon) {
  display: none;
}
```

Unfortunately, Sonar's "Selectors should be known" rule is complaining about the icon selector, which actually isn't really a selector but the value of the part attribute provided on the ion-select Shadow Dom element. Since the value of part attributes can be anything it is not possible to simply add a general exception to the rule for out whole project.

Long story short, could you please add support for the Shadow Parts standard to the rule?

**Comment by vilchik-elena** [comment]
Hi,

I believe you got the concerned rule wrong, the rule reporting for you is S4660 (`selector-pseudo-element-no-unknown`) and not `selector-type-no-unknown` as you mention in the title. 

This rule is provided by Stylelint and it seems that latest version of Stylelint fixes this problem (https://github.com/stylelint/stylelint/pull/4604). We will update the version of stylelint we use currently and your FP will disappear.

Thanks for reporting!
